### PR TITLE
Removed temperature filtering from Muon Analysis

### LINKED ISSUE #21827: log data being filtered in muon analysis
the sample logs are being filtered (t<0 does not display) when the data is loaded using muon analysis. If the same data is loaded using the load button the filter does not exist.

e.g. MUSR 67909  temp_sample

If the data is loaded via the data analysis tab (e.g. all groups in multi fitting) then the sample log is not truncated.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
The muon Analysis interface was filtering the temperature sample logs for some data and not for others. This removes the temperature filtering from all cases.

**To test:**

- Open muon analysis
- Load MUSR 67909
- Look at the Temp_sample of the sample logs (has data for t<0)
- go to the settings tab and make sure multi fitting is ticked
- Go to data analysis and some new groups will be added to the ADS - these will have sample logs with values for t<0

Fixes #21827. 

[Release Notes](/mantidproject/mantid/commit/ea58511ede53cab0481c900429ccfc3ba83d27eb) 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by AnthonyLim23 — CHANGES_REQUESTED** [review]
code looks good and works as expected. Just needs to be added to the release notes

**Review by AnthonyLim23 — APPROVED** [review]
all works

File: qt/scientific_interfaces/Muon/MuonAnalysisHelper.cpp [hunk 1]
```diff
@@ -173,10 +173,11 @@ void printRunInfo(MatrixWorkspace_sptr runWs, std::ostringstream &out) {
   out << "\nAverage Temperature: ";
   if (run.hasProperty("Temp_Sample")) {
     // Filter the temperatures by the start and end times for the run.
-    run.getProperty("Temp_Sample")->filterByTime(start, end);
+    Mantid::API::Run runTemp(run);
+    runTemp.getProperty("Temp_Sample")->filterByTime(start, end);
```
Comment (NickDraper) [inline qt/scientific_interfaces/Muon/MuonAnalysisHelper.cpp]: Can we copy just that property rather than the whole run object?

**Review by NickDraper — CHANGES_REQUESTED** [review]

**Review by NickDraper — APPROVED** [review]

File: qt/scientific_interfaces/Muon/MuonAnalysisHelper.cpp [hunk 2]
```diff
@@ -173,10 +173,12 @@ void printRunInfo(MatrixWorkspace_sptr runWs, std::ostringstream &out) {
...
   if (run.hasProperty("Temp_Sample")) {
     // Filter the temperatures by the start and end times for the run.
-    run.getProperty("Temp_Sample")->filterByTime(start, end);
-
-    // Get average of the values
-    double average = run.getPropertyAsSingleValue("Temp_Sample");
+    Mantid::Kernel::SplittingInterval time_split(start, end);
+    std::vector<Mantid::Kernel::SplittingInterval> splitVector = {time_split};
+    auto Temp_Sample = run.getProperty("Temp_Sample");
+    const auto *Temp_Sample_TimeSeries =
```
Comment (martyngigg) [inline qt/scientific_interfaces/Muon/MuonAnalysisHelper.cpp]: A couple of things:
  - variables names should be camelCase
  - we should check the result of the `dynamic_cast` for safety or we'll hit a segfault

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Review by martyngigg — APPROVED** [review]
Confirmed that the latest set of changes preserves the expected behaviour.

**Pull request merged**